### PR TITLE
build: refactor external rule tests

### LINKED ISSUE #520: Compute rules: "before all" hook for "should get a list of rules" failed
This test failed!

To configure my behavior, see [the Build Cop Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/master/packages/buildcop).

If I'm commenting on this issue too often, add the `buildcop: quiet` label and
I will stop commenting.

---

commit: b6b72ee6ed8eaef1b68d206811ea7f2b62987cd6
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/fc8fcd3c-da95-4076-baef-aa26ebd91de7), [Sponge](http://sponge2/fc8fcd3c-da95-4076-baef-aa26ebd91de7)
status: failed
<details><summary>Test output</summary><br><pre>Constraint constraints/compute.restrictLoadBalancerCreationForTypes violated for projects/long-door-651. Forwarding Rule projects/long-door-651/global/forwardingRules/gcloud-51136ce3-rule-6fc581cb of type EXTERNAL_HTTP_HTTPS is not allowed.
Error: Constraint constraints/compute.restrictLoadBalancerCreationForTypes violated for projects/long-door-651. Forwarding Rule projects/long-door-651/global/forwardingRules/gcloud-51136ce3-rule-6fc581cb of type EXTERNAL_HTTP_HTTPS is not allowed.
    at new ApiError (node_modules/@google-cloud/common/build/src/util.js:59:15)
    at Util.parseHttpRespBody (node_modules/@google-cloud/common/build/src/util.js:194:38)
    at Util.handleResp (node_modules/@google-cloud/common/build/src/util.js:135:117)
    at /workspace/node_modules/@google-cloud/common/build/src/util.js:434:22
    at onResponse (node_modules/retry-request/index.js:214:7)
    at /workspace/node_modules/teeny-request/build/src/index.js:219:13
        -> /workspace/node_modules/teeny-request/src/index.ts:325:11
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)</pre></details>

### PR DESCRIPTION
**Pull request by sofisl**
Fixes #520 

This test is currently failing because we are attempting to create an external forwarding rule, which I believe gce enforcer is blocking. So, I've tried refactoring the code so that we create an internal forwarding rule and test that.

I tried many different ways of setting up this rule, including following the how-to guides [here](https://cloud.google.com/load-balancing/docs/l7-internal/setting-up-l7-internal). I could not get them to work. The only way I was successful was by using the CLI, even after trying to convert the CLI commands into REST API. 

I know this is not the ideal solution, but I've got nothing better. Would love to get a second pair of eyes on this if possible.

### CODE REVIEW

File: system-test/compute.js [hunk 1]
```diff
@@ -693,70 +694,44 @@ describe('Compute', () => {
...
-    it('should get a list of rules', async () => {
-      const [rules] = await compute.getRules();
-      assert(rules.length > 0);
+    after(async () => {
+      const [firewalls] = await compute.getFirewalls();
+      const firewallsToDelete = firewalls
+        .filter(x => x.name.includes('network-name'))
+        .map(y => y.name);
+      for (const firewall of firewallsToDelete) {
+        await deleteFirewallRule(firewall);
```
Comment (stephenplusplus) [inline system-test/compute.js:722]: Would the method `firewall.delete()` work? (You'd have to remove the `.map(y => y.name)` part above)

File: system-test/compute.js [hunk 2]
```diff
@@ -693,70 +694,44 @@ describe('Compute', () => {
...
-        urlMap: 'global/urlMaps/' + URL_MAP_NAME,
-      });
-      await create(rule, {
-        target: 'global/targetHttpProxies/' + TARGET_PROXY_NAME,
-        portRange: '8080',
-        IPProtocol: 'TCP',
-      });
+      execSync(
+        `gcloud compute networks create ${NETWORK_NAME} --subnet-mode=custom`
+      );
```
Comment (stephenplusplus) [inline system-test/compute.js:704]:
```suggestion
      const [network, operation] = await compute.createNetwork(NETWORK_NAME, {autoCreateSubnetworks: false});
      await operation.promise();
```

^ I'm not sure that's equivalent. Just a guess from the [docs](https://cloud.google.com/compute/docs/reference/rest/v1/networks#resource): 'When set to true, the VPC network is created in auto mode. When set to false, the VPC network is created in custom mode.'

File: system-test/compute.js [hunk 3]
```diff
@@ -693,70 +694,44 @@ describe('Compute', () => {
...
+      execSync(
+        `gcloud compute networks subnets create ${SUBNETWORK_NAME} --network=${NETWORK_NAME} --range=10.0.1.0/24 --region=us-central1`
+      );
```
Comment (stephenplusplus) [inline system-test/compute.js:707]:
```suggestion
      const [subnetwork, operation] = await network.createSubnetwork(SUBNETWORK_NAME, {
        region: 'us-central1',
        range: '10.0.1.0/24'
      });
      await operation.promise();
```

File: system-test/compute.js [hunk 4]
```diff
@@ -693,70 +694,44 @@ describe('Compute', () => {
...
+      execSync(
+        `gcloud compute backend-services create ${BACKEND_SERVICE_NAME} --load-balancing-scheme=INTERNAL --region=us-central1`
+      );
```
Comment (stephenplusplus) [inline system-test/compute.js:710]:
```suggestion
      await computeRequest({
        method: 'POST',
        uri: '/regions/us-central1/backendServices',
        json: {
          name: BACKEND_SERVICE_NAME,
          loadBalancingScheme: 'INTERNAL',
        },
      });
```

File: system-test/compute.js [hunk 5]
```diff
@@ -693,70 +694,44 @@ describe('Compute', () => {
...
+      execSync(
+        `gcloud compute forwarding-rules create ${RULE_NAME} --load-balancing-scheme=INTERNAL --backend-service=${BACKEND_SERVICE_NAME} --subnet=${SUBNETWORK_NAME}  --network=${NETWORK_NAME} --region=us-central1 --ports=80-82`
+      );
```
Comment (stephenplusplus) [inline system-test/compute.js:713]:
```suggestion
      const region = compute.region('us-central1');
      const [rule, operation] = region.createRule(RULE_NAME, {
        loadBalancingScheme: 'INTERNAL',
        backendService: BACKEND_SERVICE_NAME,
        subnetwork: SUBNETWORK_NAME,
        network: NETWORK_NAME,
        ports: ['80', '81', '82'],
      });
      await operation.promise();
```

File: system-test/compute.js [hunk 6]
```diff
@@ -693,70 +694,44 @@ describe('Compute', () => {
...
+      const [firewalls] = await compute.getFirewalls();
+      const firewallsToDelete = firewalls
+        .filter(x => x.name.includes('network-name'))
+        .map(y => y.name);
+      for (const firewall of firewallsToDelete) {
+        await deleteFirewallRule(firewall);
+      }
+      await deleteForwardingRules(RULE_NAME);
+      await deleteBackendService(BACKEND_SERVICE_NAME);
+      await deleteSubnetworks(SUBNETWORK_NAME);
```
Comment (stephenplusplus) [inline system-test/compute.js:726]:
```suggestion
      const [operation] = await subnetwork.delete();
      await operation.promise();
```

File: system-test/compute.js [hunk 7]
```diff
@@ -693,70 +694,44 @@ describe('Compute', () => {
...
-      assert(rules.length > 0);
+    after(async () => {
+      const [firewalls] = await compute.getFirewalls();
+      const firewallsToDelete = firewalls
+        .filter(x => x.name.includes('network-name'))
+        .map(y => y.name);
+      for (const firewall of firewallsToDelete) {
+        await deleteFirewallRule(firewall);
+      }
+      await deleteForwardingRules(RULE_NAME);
```
Comment (stephenplusplus) [inline system-test/compute.js:724]:
```suggestion
      const [operation] = await rule.delete();
      await operation.promise():
```

**Review by stephenplusplus — COMMENTED** [review]
I went through and found our API's comparable methods to the SDK calls.

**Comment by sofisl** [comment]
@stephenplusplus, I keep getting this error, which I was getting before I just decided to use the CLI 🙂

```
Invalid value for field 'resource.subnetwork': 'gcloud-1158ae29-subnetwork-ab80d4bb'. The URL is malformed.
```

When I try to add the full URI (projects/projectId etc), I get this error:

```Error: Invalid value for field 'resource.name': 'projects/long-door-651/regions/us-central1/subnetworks/gcloud-04490660-subnetwork-3a74b658'. Must be a match of regex '(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)'```

Do you know how the name should actually be formed?

**Comment by stephenplusplus** [comment]
@sofisl I ran into similar issues, but I think I was able to get the formatting correct. Here's a diff of what I tried: https://gist.github.com/stephenplusplus/87b66993a0627e114dc296bf6ea132f5

**Review by bcoe — APPROVED** [review]
👍 contingent on dropping the `gaxios` dependency.

File: package.json [hunk 8]
```diff
@@ -57,6 +57,7 @@
     "c8": "^7.0.0",
     "codecov": "^3.0.2",
     "concat-stream": "^2.0.0",
+    "gaxios": "^4.0.1",
```
Comment (bcoe) [inline package.json:60]: I think we can drop this.

**Pull request merged**